File: backend/app/schemas/audit.py
"""Audit log read API schemas (Administration/Audit screen, spec §23)."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AuditLogPublic(BaseModel):
    id: str
    actor_id: str | None
    actor_ip: str | None
    user_agent: str | None
    action: str
    object_type: str
    object_id: str | None
    before_state: dict[str, Any] | None
    after_state: dict[str, Any] | None
    result: str
    occurred_at: datetime

File: frontend/src/types/audit.ts
export interface AuditLogEntry {
  id: string;
  actor_id: string | null;
  actor_ip: string | null;
  user_agent: string | null;
  action: string;
  object_type: string;
  object_id: string | null;
  before_state: Record<string, unknown> | null;
  after_state: Record<string, unknown> | null;
  result: string;
  occurred_at: string;
}
